inference-engine/src/mkldnn_plugin/nodes/mkldnn_matrix_nms_node.cpp

```diff
@@ -57,36 +53,11 @@ MKLDNNMatrixNmsNode::MKLDNNMatrixNmsNode(const std::shared_ptr<ngraph::Node>& op
         IE_THROW(NotImplemented) << errorMessage;
     }
 
-    errorPrefix = "MatrixNMS layer with name '" + getName() + "' ";
-    const auto matrix_nms = std::dynamic_pointer_cast<const MatrixNmsIEInternal>(op);
-
-    if (getOriginalInputsNumber() != 2)
-        IE_THROW() << errorPrefix << "has incorrect number of input edges: " << getOriginalInputsNumber();
-
-    if (getOriginalOutputsNumber() != 3)
-        IE_THROW() << errorPrefix << "has incorrect number of output edges: " << getOriginalOutputsNumber();
-
-    const SizeVector& boxes_dims = inputShapes[NMS_BOXES].getStaticDims();
-    const SizeVector& scores_dims = inputShapes[NMS_SCORES].getStaticDims();
-    if (!(boxes_dims[0] == scores_dims[0] && boxes_dims[1] == scores_dims[2])) {
-        IE_THROW() << errorPrefix << "has incompatible 'boxes' and 'scores' input dmensions";
-    }
-
-    m_numBatches = boxes_dims[0];
-    m_numBoxes = boxes_dims[1];
-    if (boxes_dims.size() != 3)
-        IE_THROW() << errorPrefix << "has unsupported 'boxes' input rank: " << boxes_dims.size();
-    if (boxes_dims[2] != 4)
-        IE_THROW() << errorPrefix << "has unsupported 'boxes' input 3rd dimension size: " << boxes_dims[2];
+    m_errorPrefix = "MatrixNMS layer with name '" + getName() + "' ";
+    if (std::dynamic_pointer_cast<const MatrixNmsIEInternal>(op))
+        m_outStaticShape = true;
```

Comment:
The model could exist two structures because our [transformation](https://github.com/openvinotoolkit/openvino/blob/2b9fedfc772101f3fb5289670d3c393a730c547f/inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp#L321):
Nms->Result: the ngraph op is actually MatrixNms
Nms->Convert->Multiply->Result: the ngraph op is actually MatrixNmsIEInternal (Currently paddle yolo model will produce this sequence)
The check 'if(std::dynamic_pointer_cast<const MatrixNmsIEInternal>(op))' means we do the convert 'MatrixNms -> MatrixNmsIEInternal'.